packages/vaadin-virtual-list/src/iron-list.js

```diff
@@ -244,16 +255,20 @@ export const PolymerIronList = Class({
    * @type {number}
    */
   get firstVisibleIndex() {
-    let idx = this._firstVisibleIndexVal;
+    var idx = this._firstVisibleIndexVal;
     if (idx == null) {
-      let physicalOffset = this._physicalTop + this._scrollOffset;
+      var physicalOffset = this._physicalTop + this._scrollOffset;
 
       idx =
         this._iterateItems(function (pidx, vidx) {
-          physicalOffset += this._physicalSizes[pidx];
+          physicalOffset += this._getPhysicalSizeIncrement(pidx);
 
           if (physicalOffset > this._scrollPosition) {
-            return vidx;
+            return this.grid ? vidx - (vidx % this._itemsPerRow) : vidx;
```

Comment:
Remove `grid` mode block.

```suggestion
            return vidx;
```